aapcs64/aapcs64.rst

```diff
@@ -521,6 +538,35 @@ Elements in a short vector are numbered such that the lowest numbered element (e
 
 A language binding may define extended types that map directly onto short vectors. Short vectors are not otherwise created spontaneously (for example because a user has declared an aggregate consisting of eight consecutive byte-sized objects).
 
+Scalable Vectors
+----------------
+
+Like a short vector (see `Short Vectors`_), a scalable vector is a
+machine type that is composed of repeated instances of one fundamental
+integral or floating-point type. The number of bytes in the vector is
+always VG×8, where VG is a runtime value determined by the execution
+environment. VG is an even integer greater than or equal to 2; the ABI
+does not define an upper bound. VG is the same for all scalable vector
+and predicate types.
+
+Each element of a scalable vector has a zero-based index. When stored
+in memory, the elements are placed in index order, so that element *N*
+comes before element *N*\ +1. The layout of each individual element
+is the same as if it were scalar. When stored in a scalable vector
+register, the least significant bit of element 0 occupies bit 0
+of the corresponding short vector register. Note that the layout of the
+vector in a scalable vector register does not depend on whether the
+system is big- or little-endian.
+
+Scalable Predicates
+-------------------
+
+A scalable predicate is a machine type that is composed of VG bytes,
+with one bit for each byte of a scalable vector.
```

Comment:
May I suggest to add an example showing a scalable vector with VGx8 bytes and the correspondent scalable predicate of VG bytes, with bytes corresponding to bits? It could help answering the question that will very likely raise when reading the specs "why is it one bit for each byte of a scalable vector". Or does this goes beyond this specs?

Reply:
Yeah, I think that's out-of-scope.  The way that the bits relate to bytes when vectors and predicates are used together is really an architecture-level thing.

Reply:
The paragraph now only talks about the link between predicates and vectors in terms of their size, to avoid the previous (unintended) implication that it was saying how each bit maps to each byte.